src/Umbraco.Core/Runtime/CoreRuntimeComponent.cs

```diff
@@ -44,6 +44,9 @@ public override void Compose(Composition composition)
             composition.Container.Register<DatabaseBuilder>();
 
             // register filesystems
+
+            composition.Container.Register<IFileSystem, MediaFileSystem>((f, wrappedFileSystem) => new MediaFileSystem(wrappedFileSystem, f.GetInstance<IContentSection>(), f.GetInstance<ILogger>()));
```

Comment:
This is most likely not covered by any tests. :/
Should look into testing/verifying all composition builders etc. later.